### ISSUE TITLE
GenerateCompiledExpressionsTempFile generates 3 empty cs files during every build/design-time build 

### ISSUE THREAD

**Issue by davkean** [Microsoft/msbuild#1646]
GenerateCompiledExpressionsTempFile generates 3 empty cs files in every single design-time and regular VS build, regardless of whether the project uses Workflow or not:

![image](https://cloud.githubusercontent.com/assets/1103906/22504849/869ace22-e8cc-11e6-8a6d-510d7f6f9e7a.png)

```
1>Target "GenerateCompiledExpressionsTempFile" in file "C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319\Microsoft.WorkflowBuildExtensions.targets":
1>  Task "WriteLinesToFile"
1>    Task Parameter:File=obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_E7A71F73-0F8D-4B9B-B56E-8E70B10BC5D3.cs
1>  Done executing task "WriteLinesToFile".
1>  Task "WriteLinesToFile"
1>    Task Parameter:File=obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_036C0B5B-1481-4323-8D20-8F5ADCB23D92.cs
1>  Done executing task "WriteLinesToFile".
1>  Task "WriteLinesToFile"
1>    Task Parameter:File=obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_5937a670-0e60-4077-877b-f7221da3dda1.cs
1>  Done executing task "WriteLinesToFile".
1>  Added Item(s): 
1>      Compile=
1>          obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_E7A71F73-0F8D-4B9B-B56E-8E70B10BC5D3.cs
1>                  CopyToOutputDirectory=Never
1>  Added Item(s): 
1>      Compile=
1>          obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_036C0B5B-1481-4323-8D20-8F5ADCB23D92.cs
1>                  CopyToOutputDirectory=Never
1>  Added Item(s): 
1>      Compile=
1>          obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_5937a670-0e60-4077-877b-f7221da3dda1.cs
1>                  CopyToOutputDirectory=Never
1>Done building target "GenerateCompiledExpressionsTempFile" in project "ConsoleApp146.csproj".
```

This is unneeded I/O and work that is not needed during design-time builds.

**Comment by jasonmalinowski** [comment]
This also causes lots of chatter during configuration switch, as we have to remove/add these to the language service because the paths depend on it. When I rewrote the Workflow expression editor support in Visual Studio 2015 it was changed that these files no longer get written to (because it used to do that in the IDE...it was terrifying) but they still must exist as the Workflow expression editor uses the file paths to communicate.

It's a fair question of why these are done in non-Workflow projects, but there's not even a good reason for them to still exist in Workflow projects either. Some further tweaking in the Workflow codebase could remove the need entirely.
